rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -528,100 +710,439 @@ def slice_state_history(
+                state_hash = self._hash_example(tracker, states)
+                state_hash_to_label_list_instances[state_hash].append(label)
+
+                example_states.append(states)
+                example_labels.append(label)
+                example_entities.append(entities)
+
+                pbar.set_postfix({f"# {self.LABEL_NAME}": f"{len(example_labels):d}"})
+
+        self._remove_user_text_if_intent(example_states)
+
+        # Collect all positive intent labels for a given state hash and add
+        # the set back to the singleton labels.
+        for positive_label_list in state_hash_to_label_list_instances.values():
+            # Get the set of positive labels associated with each state hash.
+            positive_label_set = set([labels[0] for labels in positive_label_list])
+
+            for labels in positive_label_list:
+                # Extend the singletone `labels` with the `positive_label_set`.
+                # Note that we need to filter out the redundant label `labels[0]`
+                # from `positive_label_set` so as not to add it twice.
+                filtered_label_set = filter(
+                    lambda label: label != labels[0], positive_label_set
+                )
+                labels.extend(filtered_label_set)
+
+        logger.debug(f"Created {len(example_states)} {self.LABEL_NAME} examples.")
+
+        return example_states, example_labels, example_entities
+
+    def _extract_examples(
+        self,
+        tracker: DialogueStateTracker,
+        domain: Domain,
+        omit_unset_slots: bool = False,
+        ignore_action_unlikely_intent: bool = False,
+    ) -> Iterator[Tuple[List[State], List[Text], List[Dict[Text, Any]]]]:
+        """Create an iterator over training examples from a tracker.
+
+        Create an iterator over the training examples that will be created
+        from the provided tracker.
+
+        Args:
+            trackers: The tracker from which to extract training examples.
+            domain: The domain of the training data.
+            omit_unset_slots: If `True` do not include the initial values of slots.
+            ignore_action_unlikely_intent: Remove action_unlikely_intent from
+                training states.
+
+        Returns:
+            An iterator over tuples of list of states, list of labels and list of
+                entity data.
+        """
+        tracker_states = self._create_states(
+            tracker, domain, omit_unset_slots=omit_unset_slots
+        )
+
+        if ignore_action_unlikely_intent:
+            tracker_states = self._remove_action_unlikely_intent(tracker_states)
+
+        label_index = 0
+        for event in tracker.applied_events():
+
+            if isinstance(event, ActionExecuted):
+                if (
+                    ignore_action_unlikely_intent
+                    and event.action_name == ACTION_UNLIKELY_INTENT_NAME
+                ):
+                    continue
```

Comment:
Same feedback here about `ignore_Action_unlikely_intent` and moving the slicing outside of the function to reduce the dependency between the two.